### ISSUE TITLE
Existing worktrees recognized, but incorrectly organized

### ISSUE THREAD

**Issue by socketbox** [gitkraken/vscode-gitlens#2575]
### Description

I'm a new VSCode and GitLens user, and have an existing project (in the general, non-IDE-specific sense) that uses git worktrees. I created this project in a directory  by doing the following: running `git clone --bare <repourl> .bare` and creating a `.git` file with  a `gitdir` key whose value is the relative path to `.bare`. 

The working project directory looks something like this:
```
.bare/
.git
main/
RWEB-7612-rm-confd/
RWEB-7649-restructure-tf/
RWEB-7693-docs/
RWEB-7733-nextjs-edcar/
RWEB-7793-edcar-nextjs-ci/
RWEB-7796-remove-awscli/
RWEB-7798-lean-img/
RWEB-7799-dev-creds-script/
RWEB-7845-tf-workflow/
RWEB-7853-refactor-workflows/
RWEB-7865-compose-nextjs/
```
The file `.git` contains:
```
 gitdir: ./.bare
```

There are a couple of problems I encounter immediately:

1. When I  open the project's folder in VSCode, the worktress shown under  `WORKTREES` are oddly nested:
![image](https://user-images.githubusercontent.com/1593835/225349846-d433d43c-3472-4957-8856-fb4c5781c841.png)
2. The `.bare` directory is shown with a folder icon.
3. When I click on the `+` button to add a worktree, I'm told to choose a repository, but presented with a list of my worktrees. Clicking on any of the displayed worktrees doesn't result in anything.

### GitLens Version

13.3.2

### VS Code Version

Version: 1.76.1
Commit: 5e805b79fcb6ba4c2d23712967df89a089da575b
Date: 2023-03-08T16:31:31.010Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Linux x64 5.15.0-67-generic
Sandboxed: No

### Git Version

2.40.0

### GitLens+ Worktree settings
![image](https://user-images.githubusercontent.com/1593835/225354006-dd44c7c3-48be-4dc4-8247-9373e8e6ab67.png)

**Comment by eamodio** [comment]
Can you please see if this issue is still reproducible in the pre-release edition of GitLens? Thanks!

You can switch to the pre-release edition of GitLens, by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.

![image](https://user-images.githubusercontent.com/641685/199397587-6948b1bc-3019-4d85-a6d5-332fea5d8f78.png)

**Comment by socketbox** [comment]
> Can you please see if this issue is still reproducible in the pre-release edition of GitLens? Thanks!
> 
Yes, it seems to still be a problem under pre-release (`v2023.3.1605`):
![image](https://user-images.githubusercontent.com/1593835/225633295-5a6e2756-9133-4a2d-88ca-322d0b76c32e.png)
